--- a/kernel/infra/file_wrapper.c
+++ b/kernel/infra/file_wrapper.c
@@ -432,8 +432,6 @@
 static struct inode *ksu_anon_inode_make_secure_inode(const char *name, const struct inode *context_inode)
 {
     struct inode *inode;
-    const struct qstr qname = QSTR_INIT(name, strlen(name));
-    int error;
 
     if (unlikely(!anon_inode_mnt)) {
         return ERR_PTR(-ENODEV);
@@ -444,11 +442,6 @@
     if (IS_ERR(inode))
         return inode;
     inode->i_flags &= ~S_PRIVATE;
-    error = security_inode_init_security_anon(inode, &qname, context_inode);
-    if (error) {
-        iput(inode);
-        return ERR_PTR(error);
-    }
     return inode;
 }
