--- a/kernel/runtime/ksud.c
+++ b/kernel/runtime/ksud.c
@@ -61,7 +61,9 @@ void on_post_fs_data(void)
 #else
     stop_input_hook();
 #endif
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 10, 0)
     ksu_selinux_hide_handle_post_fs_data();
+#endif
 }
 
 extern void ext4_unregister_sysfs(struct super_block *sb);
@@ -223,7 +225,9 @@ int ksu_handle_execveat_ksud(int *fd, struct filename **filename_ptr,
             char buf[16];
             if (check_argv(*argv, 1, "second_stage", buf, sizeof(buf))) {
                 pr_info("/system/bin/init second_stage executed\n");
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 10, 0)
                 ksu_selinux_hide_handle_second_stage();
+#endif
                 apply_kernelsu_rules();
                 cache_sid();
                 setup_ksu_cred();
@@ -295,7 +297,9 @@ int ksu_handle_execveat_ksud(int *fd, struct filename **filename_ptr,
         char buf[16];
         if (!init_second_stage_executed && check_argv(*argv, 1, "second_stage", buf, sizeof(buf))) {
             pr_info("/system/bin/init second_stage executed\n");
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 10, 0)
             ksu_selinux_hide_handle_second_stage();
+#endif
             apply_kernelsu_rules();
             cache_sid();
             setup_ksu_cred();
--- a/kernel/sulog/event.c
+++ b/kernel/sulog/event.c
@@ -374,7 +374,7 @@ static struct ksu_sulog_pending_event *ksu_sulog_capture_grant_root(const struct
     struct ksu_sulog_event *event;
 
     // This is actually stupid fix
-    #define USER_ARG_NULL user_arg_null_ptr()
+    #define USER_ARG_NULL (*user_arg_null_ptr())
 
     pending = ksu_sulog_capture(KSU_SULOG_EVENT_IOCTL_GRANT_ROOT, NULL, USER_ARG_NULL, gfp);
     if (!pending)
